pkg/cmd/dev/generate.go

```diff
@@ -288,70 +278,17 @@ func (d *dev) generateCgo(cmd *cobra.Command) error {
 	if err != nil {
 		return err
 	}
-
-	const cgoTmpl = `// GENERATED FILE DO NOT EDIT
-
-package {{ .Package }}
-
-// #cgo CPPFLAGS: {{ .CPPFlags }}
-// #cgo LDFLAGS: {{ .LDFlags }}
-import "C"
-`
-
-	tpl := template.Must(template.New("source").Parse(cgoTmpl))
-	archived, err := d.getArchivedCdepString(bazelBin)
-	if err != nil {
-		return err
-	}
-	// Figure out where to find the c-deps libraries.
-	var jemallocDir, projDir, krbDir string
-	if archived != "" {
-		execRoot, err := d.getExecutionRoot(ctx)
-		if err != nil {
-			return err
-		}
-		jemallocDir = filepath.Join(execRoot, "external", fmt.Sprintf("archived_cdep_libjemalloc_%s", archived))
-		projDir = filepath.Join(execRoot, "external", fmt.Sprintf("archived_cdep_libproj_%s", archived))
-		if runtime.GOOS == "linux" {
-			krbDir = filepath.Join(execRoot, "external", fmt.Sprintf("archived_cdep_libkrb5_%s", archived))
-		}
-	} else {
-		jemallocDir = filepath.Join(bazelBin, "c-deps/libjemalloc_foreign")
-		projDir = filepath.Join(bazelBin, "c-deps/libproj_foreign")
-		if runtime.GOOS == "linux" {
-			krbDir = filepath.Join(bazelBin, "c-deps/libkrb5_foreign")
-		}
-	}
-	cppFlags := fmt.Sprintf("-I%s", filepath.Join(jemallocDir, "include"))
-	ldFlags := fmt.Sprintf("-L%s -L%s", filepath.Join(jemallocDir, "lib"), filepath.Join(projDir, "lib"))
-	if krbDir != "" {
-		cppFlags += fmt.Sprintf(" -I%s", filepath.Join(krbDir, "include"))
-		ldFlags += fmt.Sprintf(" -L%s", filepath.Join(krbDir, "lib"))
-	}
-
-	cgoPkgs := []string{
-		"pkg/cli",
-		"pkg/cli/clisqlshell",
-		"pkg/server/status",
-		"pkg/ccl/gssapiccl",
-		"pkg/geo/geoproj",
+	args := []string{
+		"run",
+		"//pkg/cmd/generate-cgo:generate-cgo",
```

Comment:
Should add:

```
"--run_under=cd [workspace] && ",
```

Or something like this to make sure the binary is executed in the workspace directory.